src/main/java/com/bobocode/hoverla/bring/context/ApplicationContextImpl.java

```diff
@@ -1,60 +1,146 @@
+        // TODO:  add implementation
+    }
 
+    @Override
+    public void register(Class<?>... beanTypes) {
+        // TODO:  add implementation
     }
 
     @Override
     public <T> String register(Class<T> beanType) {
+        // TODO: add implementation
         return null;
     }
 
     @Override
     public <T> T getBean(Class<T> beanType) {
-        return null;
+        checkNotNull(beanType, "The argument [beanType] must be not null");
+
+        List<T> beans = beanDefinitionTable.column(beanType)
+                .values()
+                .stream()
+                .map(beanDefinition -> beanType.cast(beanDefinition.instance()))
+                .toList();
+        if (beans.isEmpty()) {
+            throw new NoSuchBeanException("Bean with provided name/type [%s] not found in the context".formatted(beanType.getSimpleName()));
+        } else if (beans.size() > 1) {
+            throw new NoUniqueBeanException("Expected single bean of type %s, but found %d".formatted(beanType.getSimpleName(), beans.size()));
+        } else return beans.get(0);
     }
 
     @Override
     public Object getBean(String beanName) {
-        return null;
+        checkBeanName(beanName);
+
+        return beanDefinitionTable.row(beanName)
+                .values()
+                .stream()
+                .findFirst()
+                .map(BeanDefinition::instance)
+                .orElseThrow(() -> new NoSuchBeanException("Bean with provided name/type [%s] not found in the context".formatted(beanName)));
     }
 
     @Override
     public <T> T getBean(String beanName, Class<T> beanType) {
-        return null;
+        checkNotNull(beanType, "The argument [beanType] must be not null");
+        checkBeanName(beanName);
+
+        Object bean = getBean(beanName);
+        if (bean.getClass().isAssignableFrom(beanType)) {
+            return beanType.cast(bean);
+        } else {
+            throw new NoSuchBeanException("Bean with provided name/type [%s] not found in the context".formatted(beanName));
+        }
     }
 
     @Override
     public <T> Map<String, T> getAllBeans(Class<T> beanType) {
-        return null;
+        checkNotNull(beanType, "The argument [beanType] must be not null");
+
+        return beanDefinitionTable.column(beanType)
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> beanType.cast(entry.getValue().instance())));
     }
 
     @Override
-    public boolean containsBean(String name) {
-        return false;
+    public boolean containsBean(String beanName) {
+        checkBeanName(beanName);
+        return !beanDefinitionTable.row(beanName).isEmpty();
     }
+
+    private void checkBeanName(String beanName) {
+        checkArgument(isNotEmpty(beanName), "The argument [beanName] must be not null or empty");
+        checkArgument(isNotBlank(beanName), "The argument [beanName] must be not blank");
```

Comment:
Hm, it seems to me that checking if we have any space in a name should be enough, so we don't need additional check for blank